core/dbt/context/providers.py

```diff
@@ -1432,6 +1432,20 @@ def this(self) -> Optional[RelationProxy]:
             return None
         return self.db_wrapper.Relation.create_from(self.config, self.model)
 
+    @contextproperty
+    def state_relation(self) -> Optional[RelationProxy]:
```

Comment:
Is now the time for us to actually decide what this thing should be called? (Or was that back in https://github.com/dbt-labs/dbt-core/pull/7796?) `state_relation` is the least bad thing I could come up with at the time, but it's definitely a bit wonky.

Another question: Should this even have the same name as the node attribute, given that we're actually converting it to a real (adapter-informed) `Relation` object here?

Reply:
I would prefer a narrower scoped name, we can always expand the scope later on but it's hard to narrow it. Also making names slightly more verbose sometimes makes the code way easy to read, it took me a while to make the link of `state_relation` means a relation for current model in the provided state.

With that being said, I do not have a great suggestion on what it should be named. Throwing some potentially bad options here: `source_relation`, `provided_state_relation`.

Reply:
I have strong preference against `state_relation` as it is too vague. What does `state` mean? You can even specify two previous state to dbt now.
`defer_state_relation` or something that gives reader of the code more hint would be my preference. Feel free to update it to any name that's not `state_relation`.

Reply:
I like the idea of including "defer" in the name, since `clone` should always be using the deferral state, rather than comparison state (if two are provided separately).

How about: `defer_relation` or `defer_to_relation`?